### PR TITLE
Fix build of parallel

### LINKED ISSUE #241: Build Fails with HIGHFIVE_PARALLEL_HDF5
I turned on HIGHFIVE_PARALLEL_HDF5 and now I'm getting build failures. My cmake output was 

> -- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/tcs/gcc/4.8.5/bin/gcc
-- Check for working C compiler: /usr/tcs/gcc/4.8.5/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/tcs/gcc/4.8.5/bin/g++
-- Check for working CXX compiler: /usr/tcs/gcc/4.8.5/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_CXX14
-- Performing Test COMPILER_SUPPORTS_CXX14 - Success
-- Performing Test COMPILER_SUPPORTS_CXX17
-- Performing Test COMPILER_SUPPORTS_CXX17 - Success
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /home/tee31163/anaconda3/lib/libhdf5.so;/usr/lib64/librt.so;/usr/lib64/libpthread.so;/home/tee31163/anaconda3/lib/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.10.4")
-- Boost version: 1.53.0
-- Found the following Boost libraries:
--   system
--   serialization
--   unit_test_framework
-- Found MPI_C: /usr/lib64/openmpi/lib/libmpi.so (found version "3.0")
-- Found MPI_CXX: /usr/lib64/openmpi/lib/libmpi_cxx.so (found version "3.0")
-- Found MPI: TRUE (found version "3.0")
--  - AUTO_TEST_WITH_MPIEXEC cmd /usr/lib64/openmpi/bin/mpiexec
--   -- set MPI test execution prefix to /usr/lib64/openmpi/bin/mpiexec
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components:  doxygen dot
-- Configuring done
-- Generating done
-- Build files have been written to: /home/tee31163/HighFive/build

The verbose make output is:

> /home/tee31163/anaconda3/bin/cmake -S/home/tee31163/HighFive -B/home/tee31163/HighFive/build --check-build-system CMakeFiles/Makefile.cmake 0
/home/tee31163/anaconda3/bin/cmake -E cmake_progress_start /home/tee31163/HighFive/build/CMakeFiles /home/tee31163/HighFive/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory `/home/tee31163/HighFive/build'
make -f src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/build.make src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/depend
make[2]: Entering directory `/home/tee31163/HighFive/build'
cd /home/tee31163/HighFive/build && /home/tee31163/anaconda3/bin/cmake -E cmake_depends "Unix Makefiles" /home/tee31163/HighFive /home/tee31163/HighFive/src/examples /home/tee31163/HighFive/build /home/tee31163/HighFive/build/src/examples /home/tee31163/HighFive/build/src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/DependInfo.cmake --color=
Dependee "/home/tee31163/HighFive/build/src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/DependInfo.cmake" is newer than depender "/home/tee31163/HighFive/build/src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/depend.internal".
Dependee "/home/tee31163/HighFive/build/src/examples/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/tee31163/HighFive/build/src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/depend.internal".
Scanning dependencies of target select_partial_dataset_cpp11_bin
make[2]: Leaving directory `/home/tee31163/HighFive/build'
make -f src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/build.make src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/build
make[2]: Entering directory `/home/tee31163/HighFive/build'
[  3%] Building CXX object src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/select_partial_dataset_cpp11.cpp.o
cd /home/tee31163/HighFive/build/src/examples && /usr/tcs/gcc/4.8.5/bin/g++  -DH5_USE_BOOST -DMPI_NO_CPPBIND -DNDEBUG -D_FORTIFY_SOURCE=2 -I/home/tee31163/HighFive/include -isystem /home/tee31163/anaconda3/include -isystem /usr/include/openmpi-x86_64  -g -Wall -Wextra -Werror -Wshadow -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wdouble-promotion -Wformat=2 -Wconversion -Wsign-conversion -Wno-unused-parameter -O2   -std=gnu++1y -o CMakeFiles/select_partial_dataset_cpp11_bin.dir/select_partial_dataset_cpp11.cpp.o -c /home/tee31163/HighFive/src/examples/select_partial_dataset_cpp11.cpp
[  6%] Linking CXX executable select_partial_dataset_cpp11_bin
cd /home/tee31163/HighFive/build/src/examples && /home/tee31163/anaconda3/bin/cmake -E cmake_link_script CMakeFiles/select_partial_dataset_cpp11_bin.dir/link.txt --verbose=1
/usr/tcs/gcc/4.8.5/bin/g++  -g -Wall -Wextra -Werror -Wshadow -Wnon-virtual-dtor -Wcast-align -Wunused -Woverloaded-virtual -Wpedantic -Wdouble-promotion -Wformat=2 -Wconversion -Wsign-conversion -Wno-unused-parameter -O2  "-Wl,-rpath -Wl,/usr/lib64/openmpi/lib -Wl,--enable-new-dtags -pthread" CMakeFiles/select_partial_dataset_cpp11_bin.dir/select_partial_dataset_cpp11.cpp.o  -o select_partial_dataset_cpp11_bin -Wl,-rpath,/home/tee31163/anaconda3/lib:/usr/lib64/openmpi/lib /home/tee31163/anaconda3/lib/libhdf5.so /usr/lib64/librt.so -lpthread /home/tee31163/anaconda3/lib/libz.so /usr/lib64/libdl.so /usr/lib64/libm.so /usr/lib64/openmpi/lib/libmpi.so /usr/lib64/libboost_serialization-mt.so
/usr/bin/ld: unrecognised option: -rpath -Wl
collect2: error: ld returned 1 exit status
make[2]: *** [src/examples/select_partial_dataset_cpp11_bin] Error 1
make[2]: Leaving directory `/home/tee31163/HighFive/build'
make[1]: *** [src/examples/CMakeFiles/select_partial_dataset_cpp11_bin.dir/all] Error 2
make[1]: Leaving directory `/home/tee31163/HighFive/build'
make: *** [all] Error 2

This was with a recently cloned repo. The only modification I made was turning on parallel builds. It worked before making this change.

### PR DESCRIPTION
**Pull request by alkino**
Fix #241

### CODE REVIEW

**Comment by ferdonline** [comment]
`Error: invalid option: --with-mpi`
AFAIK they brew dont have versions compiled with mpi

**Comment by alkino** [comment]
We merge like that?

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ addons: &clang6
     packages:
       - clang-6.0
       - libboost-all-dev
-      - libhdf5-serial-dev
+      - libhdf5-openmpi-dev
```
Comment (ferdonline) [inline .travis.yml]: maybe here we can leave serial test, so with clang we have one of each kind

File: .travis.yml [hunk 2]
```diff
@@ -47,26 +47,29 @@ matrix:
...
       - USE_XTENSOR=True
+      - HIGHFIVE_PARALLEL_HDF5=True
       os: linux
       dist: xenial
       addons: *gcc7
 
     # Test clang-6.0
     - env:
       - CLANG_VERSION=6.0
+      - HIGHFIVE_PARALLEL_HDF5=True
```
Comment (ferdonline) [inline .travis.yml]: In that case we have parallel false here

File: CMake/PackageConfig.cmake [hunk 3]
```diff
@@ -10,12 +10,12 @@ target_compile_definitions(HighFive INTERFACE ${HDF5_DEFINITIONS})
 
 # MPI
 if(HIGHFIVE_PARALLEL_HDF5)
-  target_include_directories(HighFive SYSTEM INTERFACE ${MPI_C_INCLUDE_PATH})
-  target_link_libraries(HighFive INTERFACE ${MPI_C_LIBRARIES})
+  target_include_directories(HighFive SYSTEM INTERFACE ${MPI_CXX_INCLUDE_PATH})
```
Comment (ferdonline) [inline CMake/PackageConfig.cmake]: Were these changes required to it to succeed? Or is it just an improvement?

**Comment by ferdonline** [comment]
We should support the scenario added (using a parallel hdf5 but compile serial apps)

**Comment by alkino** [comment]
Really good catch, I fix it but it's only that findHDF5 don't do its job well
